src/t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.hxx

```diff
@@ -0,0 +1,507 @@
+   * \return  Basis functions evaluated at the reference point.
+   */
+  std::vector<double>
+  t8_geom_h8_basis (const double *ref_point) const;
+
+  /**
+   * Basis functions of a 27-node hexahedron element.
+   * \verbatim
+          z = 0                     z = 0.5                   z = 1
+     y                         y                         y
+      ^                         ^                         ^
+      |                         |                         |
+     2     21     3            8     23      14          6     22      7
+      x ----x---- x             x ----x---- x             x ----x---- x
+      |           |             |           |             |           |
+      |    24     |             |    26     |             |    25     |
+    10x     x     x15         12x     x     x17         11x     x     x16
+      |           |             |           |             |           |
+      |           |    x        |           |    x        |           |    x
+      x ----x---- x  -->        x ----x---- x  -->        x ----x---- x  -->
+     0     18     1            9     20      13          4     19      5
+     \endverbatim
+   * \param ref_point  Point in the reference space.
+   * \return  Basis functions evaluated at the reference point.
+   */
+  std::vector<double>
+  t8_geom_h27_basis (const double *ref_point) const;
+
+  /** Polynomial degree of the interpolation. */
+  const int *degree;
+};
+
+/**
+ * Flatten a vector of vector into a single vector.
+ * 
+ * \tparam T   Template parameter of a vector.
+ * \param vec  Nested vector to be flattened.
+ * \return     Flattened vector.
+ */
+template <typename T>
+std::vector<T>
+flatten (const std::vector<std::vector<T>> &vec)
+{
+  std::vector<T> flattened;
+  for (auto const &v : vec) {
+    flattened.insert (flattened.end (), v.begin (), v.end ());
+  }
+  return flattened;
+}
+
+/**
+ * A single coarse mesh cell with Lagrange geometry.
+ * 
+ * This class is essentially a wrapper around a cmesh.
+ * By having a single element instead of a mesh, understanding and
+ * testing is made easier. Several topological utilities are provided,
+ * some specific to the Lagrange geometry, some valid for all the
+ * geometries in t8code.
+ */
+class LagrangeElement {
+ public:
+  /**
+   * Construct a new LagrangeElement object.
+   * 
+   * \param eclass  Element class (line, quad, etc.)
+   * \param degree  Polynomial degree (1, 2, ...)
+   * \param nodes   x,y,z coordinates of the nodes, adhering to the numbering
+   *                convention.
+   */
+  LagrangeElement (t8_eclass_t eclass, uint degree, std::vector<double> &nodes);
+
+  /**
+   * Destroy the LagrangeElement object.
+   * 
+   * The cmesh wrapped by this class is also destroyed.
+   * 
+   */
+  ~LagrangeElement ()
+  {
+    t8_cmesh_destroy (&cmesh);
+  };
+
+  /**
+   * Get the type of the element.
+   * 
+   * \return  Element class of the element.
+   */
+  t8_eclass_t
+  getType () const;
```

Comment:
Please adhere to the naming conventions. But we decided, that member functions do not have to start with t8_

Reply:
So will the methods be `get_type`, `face_classes`, etc. (i.e. snake case naming)?

Reply:
Yes exactly